desc/grid.py

```diff
@@ -447,34 +450,75 @@ def _create_nodes(
             if self.sym:
                 t += t[1] / 2
             dt = 2 * np.pi / t.size * np.ones_like(t)
+            if endpoint and t.size > 2:
+                # When endpoint is True, the theta = 0 node is duplicated at 2 pi.
```

Comment:
I think we already do this in `_scale_weights`, should double check that we're not doing it twice

Reply:
yeah the grids for surfaces and curves use `endpoint=True`, so that's probably what's causing the failing test

Reply:
Yes I have fixed that